lib/internal/Magento/Framework/CompiledInterception/Generator/CompiledInterceptor.php

```diff
@@ -0,0 +1,532 @@
+        } else {
+            $body = [];
+        }
+
+        $resultChain = [];
+        if (isset($conf[DefinitionInterface::LISTENER_AROUND])) {
+            $resultChain[] = $this->compileAroundPlugin($methodName, $conf[DefinitionInterface::LISTENER_AROUND],  $capitalizedName, $extraParams, $parameters, $returnVoid);
+        } else {
+            $resultChain[] = ["parent::{$methodName}({$this->getParameterList($parameters)});"];
+        }
+
+        if (isset($conf[DefinitionInterface::LISTENER_AFTER])) {
+            $resultChain = array_merge($resultChain, $this->compileAfterPlugins($conf[DefinitionInterface::LISTENER_AFTER], 'after' . $capitalizedName, $extraParams, $returnVoid));
+        }
+        foreach ($resultChain as $lp => $piece) {
+            if ($first) $first = false; else $body[] = "";
+            if (!$returnVoid) {
+                $piece[0] = (($lp + 1 == count($resultChain)) ? "return " : "\$result = ") . $piece[0];
+            }
+            foreach ($piece as $line) {
+                $body[] = $line;
+            }
+        }
+        return $body;
+    }
+
+    /**
+     * @param array $parameters
+     * @return string
+     */
+    private function getParameterListForNextCallback(array $parameters)
+    {
+        $ret = [];
+        foreach ($parameters as $parameter) {
+            $ret [] =
+                ($parameter->isPassedByReference() ? '&' : '') .
+                "\${$parameter->getName()}" .
+                ($parameter->isDefaultValueAvailable() ?
+                    ' = ' . ($parameter->isDefaultValueConstant() ?
+                        $parameter->getDefaultValueConstantName() :
+                        str_replace("\n", '', var_export($parameter->getDefaultValue(), true))) :
+                    '');
+        }
+        return implode(', ', $ret);
+    }
+
+    /**
+     * @param \ReflectionParameter[]  $parameters
+     * @return string
+     */
+    private function getParameterList(array $parameters)
+    {
+        $ret = [];
+        foreach ($parameters as $parameter) {
+            $ret [] = "\${$parameter->getName()}";
+        }
+        return implode(', ', $ret);
+    }
+
+    /**
+     * @param $plugin
+     * @return string
+     */
+    private function getGetterName($plugin)
+    {
+        return '____plugin_' . $plugin['clean_name'];
+    }
+
+    /**
+     * @param $plugin
+     * @return array
+     */
+    private function getPluginPropertyInfo($plugin)
+    {
+        return [
+            'name' => '____plugin_' . $plugin['clean_name'],
+            'visibility' => 'private',
+            'docblock' => [
+                'tags' => [['name' => 'var', 'description' => '\\' . $plugin['class']]],
+            ]
+        ];
+    }
+
+    /**
+     * @param $plugin
+     * @return array
+     */
+    private function getPluginGetterInfo($plugin)
```

Comment:
When I bench-marked this, using property gave a tiny speed benefit. 
Also without this property this getter is moot because `ObjectManager` could be used inline.
Isn't this getter with a cleanly defined return type readable enough?

Reply:
The method by itself is readable enough (if not to take into account the method name and FQCN for return type) but the method body looks noisy, the properties names too long.